code/modules/vehicles/mecha/equipment/tools/medical_tools.dm

```diff
@@ -283,23 +278,55 @@
 	return ..()
 
 /obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/get_snowflake_data()
-	return list(
+	var/list/analyzed_reagents = list() // we need to make this list because .tsk wont map over an indexed array
+	for(var/i=1 to known_reagents.len)
+		var/enabled = FALSE
+		if(known_reagents[i] in processed_reagents)
+			enabled = TRUE
+		analyzed_reagents += list((list("name" = known_reagents[i].name, "enabled" = enabled)))
+	var/list/data = list(
 		"snowflake_id" = MECHA_SNOWFLAKE_ID_SYRINGE,
 		"mode" = mode == FIRE_SYRINGE_MODE ? "Launch" : "Analyze",
 		"mode_label" = "Action",
 		"syringe" = LAZYLEN(syringes),
 		"max_syringe" = max_syringes,
 		"reagents" = reagents.total_volume,
 		"total_reagents" = reagents.maximum_volume,
+		"analyzed_reagents" = analyzed_reagents,
 	)
+	var/list/contained_reagents = list()
+	if(length(reagents.reagent_list))
+		for(var/datum/reagent/reagent in reagents.reagent_list)
```

Comment:
```suggestion
		for(var/datum/reagent/reagent as anything in reagents.reagent_list)
```

don't do this if `reagent_list` is not guaranteed to have stuff that's not `/datum/reagent` though.